core/src/main/java/com/dsingley/jwt/core/SigningAlgorithmSupplier.java

```diff
@@ -0,0 +1,8 @@
+package com.dsingley.jwt.core;
+
+import com.auth0.jwt.algorithms.Algorithm;
+
+public interface SigningAlgorithmSupplier {
```

Comment:
You _could_ just use a regular `java.util.Supplier` instead of this...since it's basically just a `Supplier` with a predefined type. I suppose you could also _extend_ `Supplier`, something like:

```java
public interface SigningAlgorithmSupplier extends Supplier<Algorithm> {
}
```

and you could add a `getAlgoritm()` if you want, or just use the default `get` method.